### PR TITLE
Matched func_802BD758_3184A8 and Matched func_801363A0

### PR DESCRIPTION
**Pull request by MS-DOS1999**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀 This PR matches 2 functions (+0.12%) on `us`.

File: src/C50A0.c [hunk 1]
```diff
@@ -2795,25 +2794,26 @@ void func_801363A0(ItemEntity* itemEntity) {
...
     s32 temp2;
     s32 v1;
+    u32 flags;
 
     switch (itemEntity->state) {
         case 2:
         case 10:
             if (!(itemData->typeFlags & ITEM_TYPE_FLAG_BADGE)) {
                 if (!(itemEntity->flags & ITEM_ENTITY_FLAGS_4000000) || (itemEntity->pickupMsgFlags & 0x4)) {
-                    itemMsg = MSG_Menus_0058;
```
Comment (HailSanta) [inline src/C50A0.c:2804]: These should not have been reverted to MESSAGE_ID, please use MSG_*

File: src/C50A0.c [hunk 2]
```diff
@@ -2857,37 +2857,39 @@ void func_801363A0(ItemEntity* itemEntity) {
...
+
+            if (!(gItemTable[itemEntity->itemID].typeFlags & ITEM_TYPE_FLAG_KEY) &&
+                    itemEntity->itemID != ITEM_STAR_PIECE &&
+                    !(gItemTable[itemEntity->itemID].typeFlags & ITEM_TYPE_FLAG_GEAR) &&
+                    !(itemEntity->pickupMsgFlags & 0x30)) {
+                set_window_properties(0xC, s1, s3 - 0x18 + temp, offsetY,
                                     temp2, 0, func_8013673C, itemEntity, -1);
             } else if (gItemTable[itemEntity->itemID].typeFlags & 1){
-                set_window_properties(WINDOW_ID_12, s1, s3 - 0x18 + temp, offsetY,
+                set_window_properties(0xC, s1, s3 - 0x18 + temp, offsetY,
```
Comment (HailSanta) [inline src/C50A0.c:2868]: WINDOW_ID_*

File: src/world/partner/bombette.c [hunk 3]
```diff
@@ -237,101 +241,103 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
-                break;
-            }
+        if (playerStatus->flags & 2) {
+            evt->functionTemp[0] = 7;
+            break;
+        }
+        npc->moveToPos.x = playerStatus->position.x;
+        npc->moveToPos.y = playerStatus->position.y;
+        npc->moveToPos.z = playerStatus->position.z;
+        npc->currentAnim = 0x30007;
```
Comment (HailSanta) [inline src/world/partner/bombette.c:263]: ANIM_WorldBombette_Run

File: src/world/partner/bombette.c [hunk 4]
```diff
@@ -195,34 +195,38 @@ s32 world_bombette_can_player_pause(Npc* npc) {
...
         func_802BD6DC_31842C(npc);
         evt->functionTemp[0] = 20;
     }
-
     switch (evt->functionTemp[0]) {
         case 20:
-            if ((playerStatus->inputEnabledCounter != 0) || (playerStatus->flags & 2) || !(npc->flags & 0x1000)) {
-                return ApiStatus_DONE2;
+            if (((playerStatus->inputEnabledCounter != 0) || (playerStatus->flags & 2)) || (!(npc->flags & 0x1000))) {
+                return 2;
```
Comment (HailSanta) [inline src/world/partner/bombette.c:229]: Same issues in this file with enum values.

**Review by HailSanta — CHANGES_REQUESTED** [review]

File: src/world/partner/bombette.c [hunk 5]
```diff
@@ -237,101 +241,103 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
-                suggest_player_anim_clearUnkFlag(ANIM_Mario_6000C);
-                npc->yaw = playerStatus->targetYaw;
-                npc->currentAnim = ANIM_WorldBombette_Walk;
-                evt->functionTemp[0] = 1;
-                evt->functionTemp[1] = 10;
-            }
         case 1:
-            if (playerStatus->actionState == 0x15 || playerStatus->actionState == 0x16) {
-                evt->functionTemp[1] = 7;
+            if ((playerStatus->actionState == 0x15) || (playerStatus->actionState == 0x16)) {
```
Comment (ethteck) [inline src/world/partner/bombette.c:293]: here 2

File: src/world/partner/bombette.c [hunk 6]
```diff
@@ -237,101 +241,103 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
+            if ((playerStatus->animFlags & 4) || ((playerStatus->actionState == 0x15) || (playerStatus->actionState == 0x16))) {
                 evt->functionTemp[0] = 7;
+jump_438:
                 break;
-            }
 
+            }
             if (evt->functionTemp[1] < 45) {
-                if (!(npc->flags & 0x2000) && (D_802BE938 == 0)) {
+                if ((!(npc->flags & 0x2000)) && (D_802BE938 == 0)) {
```
Comment (ethteck) [inline src/world/partner/bombette.c:325]: here 3

File: src/world/partner/bombette.c [hunk 7]
```diff
@@ -444,29 +448,30 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
             evt->functionTemp[1] = 10;
             evt->functionTemp[0] = 5;
             break;
+
         case 5:
             npc->pos.y += npc->jumpVelocity;
             npc->jumpVelocity -= npc->jumpScale;
-            npc->rotation.z -= ((evt->functionTemp[2] * 0x4F) / 2);
-            npc->rotation.x -= ((evt->functionTemp[2] * 0x43) / 2);
+            npc->rotation.z -= (evt->functionTemp[2] * 0x4F) / 2;
```
Comment (ethteck) [inline src/world/partner/bombette.c:455]: these rotation values should be in decimal instead of hex actually

File: src/world/partner/bombette.c [hunk 8]
```diff
@@ -478,19 +483,19 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
             }
             npc->pos.y += npc->jumpVelocity;
             npc->jumpVelocity -= npc->jumpScale;
-            if (npc->jumpVelocity < -8.0) {
+            if (npc->jumpVelocity < (-8.0)) {
                 npc->jumpVelocity = -8.0f;
             }
-            npc->rotation.z -= ((evt->functionTemp[2] * 0x4F) / 2);
-            npc->rotation.x -= ((evt->functionTemp[2] * 0x43) / 2);
+            npc->rotation.z -= (evt->functionTemp[2] * 0x4F) / 2;
```
Comment (ethteck) [inline src/world/partner/bombette.c:489]: here as well

File: src/world/partner/bombette.c [hunk 9]
```diff
@@ -444,29 +448,30 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
             npc->jumpVelocity = 0.0f;
-            npc->currentAnim = ANIM_WorldBombette_Aftermath;
+            npc->currentAnim = 0x30011;
             npc->flags |= 0x800;
             evt->functionTemp[0] = 6;
             break;
+
         case 6:
-            if (npc->pos.y + 10.0f < playerStatus->position.y + playerStatus->colliderHeight) {
+            if ((npc->pos.y + 10.0f) < (playerStatus->position.y + playerStatus->colliderHeight)) {
```
Comment (ethteck) [inline src/world/partner/bombette.c:474]: parens

File: src/world/partner/bombette.c [hunk 10]
```diff
@@ -478,19 +483,19 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
             }
             npc->pos.y += npc->jumpVelocity;
             npc->jumpVelocity -= npc->jumpScale;
-            if (npc->jumpVelocity < -8.0) {
+            if (npc->jumpVelocity < (-8.0)) {
```
Comment (ethteck) [inline src/world/partner/bombette.c:486]: parens

File: src/world/partner/bombette.c [hunk 11]
```diff
@@ -536,22 +542,25 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
+            if (D_802BE924 != 0) {
+                D_802BE924 = 0;
+                sfx_stop_sound(0x80000000);
             }
-            D_802BE924 = 0;
-            sfx_stop_sound(SOUND_80000000);
-            return ApiStatus_DONE2;
+            temp_ret = 2;
+            break;
+
```
Comment (ethteck) [inline src/world/partner/bombette.c:556]: remove empty line

File: src/world/partner/bombette.c [hunk 12]
```diff
@@ -504,14 +509,15 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
+            if (D_802BE924 != 0) {
+                D_802BE924 = 0;
+                sfx_stop_sound(0x80000000);
             }
-            sfx_stop_sound(SOUND_80000000); // TODO swapping these causes tail merging that we don't want
-            D_802BE924 = 0;
-            return ApiStatus_DONE2;
+            temp_ret = 2;
+            return temp_ret;
+
```
Comment (ethteck) [inline src/world/partner/bombette.c:520]: remove empty line

**Review by ethteck — CHANGES_REQUESTED** [review]
Great job on these!!! Just some style changes

File: src/C50A0.c [hunk 13]
```diff
@@ -2868,6 +2871,8 @@ void func_801363A0(ItemEntity* itemEntity) {
                 set_window_properties(WINDOW_ID_12, s1, s3 - 0x18 + temp, offsetY,
                                     temp2, 0, func_8013673C, itemEntity, -1);
             }
+
+
```
Comment (ethteck) [inline src/C50A0.c:2875]: no need for two empty lines in a row here

File: src/C50A0.c [hunk 14]
```diff
@@ -2879,15 +2884,12 @@ void func_801363A0(ItemEntity* itemEntity) {
             break;
         case 12:
             set_message_msg(itemData->nameMsg, 0);
-            offsetY = get_msg_width(MSG_Menus_005F, 0) + 0x36;
+            offsetY = get_msg_width(MESSAGE_ID(0x1D, 0x05F), 0) + 0x36;
```
Comment (ethteck) [inline src/C50A0.c:2887]: this message ID should still be what it used to be

File: src/world/partner/bombette.c [hunk 15]
```diff
@@ -239,81 +242,83 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
             evt->functionTemp[0] = 21;
-        case 21:
-            if (playerStatus->actionState == 0x15 || playerStatus->actionState == 0x16) {
-                disable_npc_blur(npc);
-                evt->functionTemp[0] = 7;
-                break;
-            }
 
-            if ((playerStatus->flags & 2)) {
+        case 21: {
```
Comment (ethteck) [inline src/world/partner/bombette.c:246]: Is this manual scope at the beginning of case 21 necessary? It seems strange and shouldn't be

File: src/world/partner/bombette.c [hunk 16]
```diff
@@ -355,28 +359,25 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
+                    if (evt->functionTemp[1] != 0) {
+                        evt->functionTemp[1]--;
+                        break;
                     }
                 }
-
-                npc->currentAnim = ANIM_WorldBombette_AboutToExplode;
-                evt->functionTemp[1] = 20;
+                npc->currentAnim = ANIM_WorldBombette_Still | 0x10;
+                evt->functionTemp[1] = 0x14;
```
Comment (ethteck) [inline src/world/partner/bombette.c:374]: 20 instead of 0x14

File: src/world/partner/bombette.c [hunk 17]
```diff
@@ -355,28 +359,25 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
                 evt->functionTemp[0] = 3;
                 if (playerStatus->actionState == 0) {
                     suggest_player_anim_clearUnkFlag(ANIM_Mario_10002);
                 }
             }
+
```
Comment (ethteck) [inline src/world/partner/bombette.c:380]: No need for an empty line here

File: src/world/partner/bombette.c [hunk 18]
```diff
@@ -385,23 +386,24 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
                 evt->functionTemp[1]--;
                 break;
             }
-
             if (D_802BE924 != 0) {
                 D_802BE924 = 0;
-                sfx_stop_sound(SOUND_80000000);
+                sfx_stop_sound(0x80000000);
```
Comment (ethteck) [inline src/world/partner/bombette.c:391]: Please keep the sound IDs here and below like they were

File: src/world/partner/bombette.c [hunk 19]
```diff
@@ -416,25 +418,25 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
             evt->functionTemp[1] = 3;
             evt->functionTemp[0]++;
             break;
+
```
Comment (ethteck) [inline src/world/partner/bombette.c:421]: Remove empty line after break

File: src/world/partner/bombette.c [hunk 20]
```diff
@@ -416,25 +418,25 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
             npc->jumpScale = 0.8f;
             npc->pos.y += 10.0f;
-            npc->moveToPos.y = npc->pos.y ;
+            npc->moveToPos.y = npc->pos.y;
             collisionStatus->bombetteExploded = -1;
             collisionStatus->bombetteExplosionPos.x = npc->pos.x;
             collisionStatus->bombetteExplosionPos.y = npc->pos.y;
             collisionStatus->bombetteExplosionPos.z = npc->pos.z;
-            npc->currentAnim = ANIM_WorldBombette_Aftermath;
+            npc->currentAnim = 0x30011;
```
Comment (ethteck) [inline src/world/partner/bombette.c:439]: enum value

File: src/world/partner/bombette.c [hunk 21]
```diff
@@ -504,14 +507,14 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
             npc->pos.y = playerStatus->position.y;
             npc->rotation.x = 0.0f;
             npc->rotation.z = 0.0f;
-            npc->currentAnim = ANIM_WorldBombette_Idle;
+            npc->currentAnim = 0x30003;
```
Comment (ethteck) [inline src/world/partner/bombette.c:510]: anim enum value + sound enum value below

File: src/world/partner/bombette.c [hunk 22]
```diff
@@ -523,7 +526,7 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
```
Comment (ethteck) [inline src/world/partner/bombette.c:529]: anim val, sound val below

File: src/C50A0.c [hunk 23]
```diff
@@ -2795,6 +2794,7 @@ void func_801363A0(ItemEntity* itemEntity) {
     s32 s3;
     s32 temp2;
     s32 v1;
+    u32 flags;
```
Comment (ethteck) [inline src/C50A0.c:2797]: Is this var necessary? It seems unused

File: src/world/partner/bombette.c [hunk 24]
```diff
@@ -240,80 +246,74 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
             }
-
             sfx_play_sound_at_npc(SOUND_80000000, 0, -4);
             D_802BE924 = 1;
             add_vec2D_polar(&npc->pos.x, &npc->pos.z, 0.0f, npc->yaw);
             npc->currentAnim = ANIM_WorldBombette_WalkLit;
             npc->jumpVelocity = 0.0f;
             D_802BE938 = 0;
-            npc->flags = (npc->flags | 0x200) & ~0x100;
+            npc->flags = (npc->flags | 0x200) & (~0x100);
```
Comment (ethteck) [inline src/world/partner/bombette.c:307]: these should be two separate lines ( flags |= 0x200, flags &= ~0x100)
and they should use the npc flag enum values

File: src/world/partner/bombette.c [hunk 25]
```diff
@@ -240,80 +246,74 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
...
             npc->moveSpeed = 1.0f;
             evt->functionTemp[0] = 2;
             evt->functionTemp[1] = 50;
         case 2:
-            if ((playerStatus->animFlags & 4) || (playerStatus->actionState == 0x15 || playerStatus->actionState == 0x16)) {
+            if ((playerStatus->animFlags & 4) || ((playerStatus->actionState == 0x15) || (playerStatus->actionState == 0x16))) {
```
Comment (ethteck) [inline src/world/partner/bombette.c]: Pls use the action state enum values here

File: src/world/partner/bombette.c [hunk 26]
```diff
@@ -385,25 +380,25 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
                 evt->functionTemp[1]--;
                 break;
             }
-
```
Comment (ethteck) [inline src/world/partner/bombette.c:388]: All of the line spacing changes between here and 401 should ideally be discarded

**Review by ethteck — CHANGES_REQUESTED** [review]
This is the last time, I promise

**Pull request merged**